### PR TITLE
Week 06 day 03 module 01

### PR DESCRIPTION
**Pull request by eerwitt**
Changed around a bit from 22-23, flow is roughly the same. Seems like a good intro to me.

@cameronjacoby should I cover more theoretical stuff about class vs prototype based or skip it?

### CODE REVIEW

File: week-06-intro-ruby-on-rails/day-03/module-01/readme.md [hunk 1]
```diff
@@ -1 +1,187 @@
...
+
+*Hint:* use the `#is_a?` method
+
+### Class Inheritance Tree
+
+![Class inheritance](http://i.stack.imgur.com/rvcEi.png)
+
+##Classes
+
+**Challenge:**
```
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-03/module-01/readme.md]: Can you make one of those drop-down arrows here with the solution to this "challenge". Just so they have the code example to refer back to in the notes.

File: week-06-intro-ruby-on-rails/day-03/module-01/readme.md [hunk 2]
```diff
@@ -1 +1,187 @@
...
+
+Goal: Let's create a Car that goes "Putt-Putt" when it's first *initialized*
+
+*Refresher: Classes are datatypes used to create more data. They are analogous to constructors in JavaScript.*
+
+## Attributes
+
+What should we do if we want to set attributes on the car, such as a paint color and year?
+
+**Challenge:**
```
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-03/module-01/readme.md]: Drop-down here

File: week-06-intro-ruby-on-rails/day-03/module-01/readme.md [hunk 3]
```diff
@@ -1 +1,187 @@
...
+=> "hot pink"
+```
+
+*Hint*: Use `attr_accessor`.
+
+*Bonus*: Don't use `attr_accessor`.
+
+## Methods
+
+**Challenge:**
```
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-03/module-01/readme.md]: Drop-down here

File: week-06-intro-ruby-on-rails/day-03/module-01/readme.md [hunk 4]
```diff
@@ -1 +1,187 @@
...
+=> false
+alpha.color?("orange")
+=> true
+```
+
+*Hint*: define a new method with the `def` keyword inside the class.
+
+## Initialization and Instance Variables
+
+**Challenge:**
```
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-03/module-01/readme.md]: Drop-down here

File: week-06-intro-ruby-on-rails/day-03/module-01/readme.md [hunk 5]
```diff
@@ -1 +1,187 @@
...
+lambo = Car.new("red")
+lambo.color
+=> "red"
+```
+
+*Hint*: Create an instance variable with `@`.
+
+## Class Variables
+
+What if I wanted a running counter for all the cars I've ever created? Enable the code: `Car.total` which outputs the total number of all cars ever created.
```
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-03/module-01/readme.md]: Drop-down here

File: week-06-intro-ruby-on-rails/day-03/module-01/readme.md [hunk 6]
```diff
@@ -1 +1,187 @@
...
+  def initialize
+    @speed = 0
+  end
+  def accelerate(speed)
+    @speed += speed
+  end
+end
+```
+
+**Challenge:**
```
Comment (cameronjacoby) [inline week-06-intro-ruby-on-rails/day-03/module-01/readme.md]: Drop-down here

**Comment by cameronjacoby** [comment]
@eerwitt Looks good to me! I had a few comments about adding code samples in the drop-down format to server as solutions for the "mini-challenges" throughout.

And yes, I think you should cover the theoretical stuff (class vs prototype-based). They don't _need_ to know it, but it's only going to help them in interviews and in understanding differences between programming languages. Maybe spend 10 minutes talking about it? It's not something you necessarily have to add to the notes.

**Comment by eerwitt** [comment]
@cameronjacoby updated

**Pull request merged**